speech/snippets/transcribe_batch_multiple_files_v2.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+
+# [START speech_transcribe_batch_multiple_files_v2]
+import re
+from typing import List
+
+from google.cloud import storage
+from google.cloud.speech_v2 import SpeechClient
+from google.cloud.speech_v2.types import cloud_speech
+
+
+def transcribe_batch_multiple_files_v2(
+    project_id: str,
+    gcs_uris: List[str],
+    gcs_output_path: str,
+) -> cloud_speech.BatchRecognizeResponse:
+    """Transcribes audio from a Google Cloud Storage URI.
+
+    Args:
+        project_id: The GCP project ID.
+        gcs_uris: The Google Cloud Storage URIs to transcribe.
+        gcs_output_path: The Cloud Storage URI to which to write the transcript.
+
+    Returns:
+        The BatchRecognizeResponse message.
+    """
+    # Instantiates a client
+    client = SpeechClient()
+
+    config = cloud_speech.RecognitionConfig(
+        auto_decoding_config=cloud_speech.AutoDetectDecodingConfig(),
+        language_codes=["en-US"],
+        model="long",
+    )
+
+    files = []
+    for uri in gcs_uris:
+        files.append(cloud_speech.BatchRecognizeFileMetadata(uri=uri))
```

Comment:
Use list comprehension for a more Pythonic approach
```suggestion
    files = [
        cloud_speech.BatchRecognizeFileMetadata(uri=uri)
        for uri in gcs_uris
    ]
```